File: 47주차/프로세스.cpp
#include <string>
#include <vector>
#include <deque>
using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    vector<int> res;
    deque<pair<int,int>> q;
    
    for (int i =0 ; i< priorities.size(); i++){
        q.push_back({priorities[i],i});
    }
    int cnt = 0;
    
    while (q.size()> 0) {
        pair<int,int> now = q.front();
        q.pop_front();
        int flag = true;
        for (int i =0; i < q.size(); i++) {
            if (now.first < q[i].first ) {
                q.push_back(now);
                flag = false;
                break;
            }
        }
        if (flag) res.push_back(now.second);
      
    }
    
    for (int i =0; i< res.size();i++){
        if (res[i]== location) return i+1;
    }
    
    return answer;
}

File: 66주차/프로세스.js
function solution(priorities, location) {
  let answer = 1;
  const arr = priorities.map((v, i) => ({ p: v, idx: i }));

  while (arr.length > 0) {
    const tmp = arr.shift();
    if (arr.some((value) => value.p > tmp.p)) {
      // 하나라도 우선순위가 더 높은 것이 있다면. 다시 큐에 넣기.
      arr.push(tmp);
    } else {
      if (tmp.idx === location) return answer; // 현재 처리되는 프로세스가 찾고자하는 프로세스라면 반환.
      answer += 1;
    }
  }
}
